edx/edx-ora2
# Implement functions to clone rubric into API

**TL;DR -** Add backend functions for replacing ORA rubric editor data with data from another rubric.

JIRA: [EDUCATOR-5754](https://openedx.atlassian.net/browse/EDUCATOR-5754)

**What changed?**

- Adds a method to `oa_server.js` for getting rubric data from ORA backend
- Adds methods in `oa_edit_rubric.js` to clear and replace rubric data with returned data from backend
- Adds a method in `oa_edit_listeners.js` for clearing learner training examples when a rubric is replaced
- **TEMP:** expose a dev API to the window for test invocation (re-enable by applying [this diff](https://github.com/edx/edx-ora2/files/6704632/rubric_dev_hook.diff.txt))

**What didn't change?** 
I did not add tests for the following, since unit tests either do not exist or do not cover appropriate areas in these files:
  - `oa_edit_container_item.js`
  - `oa_edit.js`
  - `oa_edit_fields.js` (there is a TODO already marked here)

**Developer Checklist**


**Testing Instructions**

Testing is done through an exposed dev API in the browser.

1. Setup an ORA with a non-default rubric. Play with different numbers of criteria/options.
2. Get the ORA location ID by hooking/logging `other_openassessments_in_course` from Studio context
(note: this is different for published and non-published ORAs and can change when an ORA is updated)
3. Create a new ORA in Studio and click the "Edit" button to open the editor window
4. Open the browser developer console and enter the following code `window.__dev__.cloneRubric(<rubric-location>)`
5. Verify that rubric is replaced with new data
6. Verify that the learner training examples are cleared

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
